### ISSUE TITLE
False positive: E1135, unsupported-membership-test when using the "before" member resulting from a pexpect spawn call

### ISSUE THREAD

**Issue by alangefe** [PyCQA/pylint#7140]
### Bug description

Pylint's type checker has incorrectly detected that I cannot use a membership test on the `before` member of the return object from a pexpect `spawn()` call.

I have included a small sample of code that demonstrates the problem.

There is a case where pylint is correct. If I call `spawn()` without specifying the encoding, the resulting object's `before` member will be `bytes`, not a `str`, and it is correct that I cannot use a membership test to check for a `str` in a `bytes` object. That is represented by this block of code. Note how I must use `decode()` before the membership test:

```
from pexpect import spawn, EOF

encoding_none_result = spawn("ls")
encoding_none_result.expect(EOF)
print("test_case.py" in encoding_none_result.before.decode("utf-8"))
```

On the other hand, pylint is wrong when I pass `encoding="utf-8"` to `spawn()`. In that case, the `before` member is a `str` and I can perform the membership test without issue:

```
from pexpect import spawn, EOF

encoding_utf8_result = spawn("ls", encoding="utf-8")
encoding_utf8_result.expect(EOF)
print("test_case.py" in encoding_utf8_result.before)
```

OS: Ubuntu 22.04
Python: python 3.10
pexpect==4.8.0
pylint==2.14.4
astroid==2.11.6

### Command used

```shell
pylint -E test_case.py
```

### Pylint output

```shell
************* Module test_case
test_case.py:9:24: E1135: Value 'encoding_utf8_result.before' doesn't support membership test (unsupported-membership-test)
```

### Expected behavior

I expect pylint to detect no problems with the block of code when `encoding="utf-8"` is passed to `spawn()`.

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
```

### OS / Environment

OS: Ubuntu 22.04

### Additional dependencies

Additional dependencies:

```
pexpect==4.8.0
```
